service/controller/v12/resource/hostedzone/resource.go

```diff
@@ -0,0 +1,154 @@
+package hostedzone
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	controllercontext "github.com/giantswarm/aws-operator/service/controller/v12/context"
+	"github.com/giantswarm/aws-operator/service/controller/v12/key"
+)
+
+const (
+	name = "hostedzonev12"
+)
+
+type Config struct {
+	HostRoute53 *route53.Route53
+	Logger      micrologger.Logger
+}
+
+type Resource struct {
+	hostRoute53 *route53.Route53
+	logger      micrologger.Logger
+}
+
+func New(config Config) (*Resource, error) {
+	if config.HostRoute53 == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.HostRoute53 must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	r := &Resource{
+		hostRoute53: config.HostRoute53,
+		logger:      config.Logger,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return name
+}
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	err := r.setStatus(ctx, obj)
```

Comment:
We will need that information on both update and delete events, so there is a private method extracted.